Pilot Data Exploration.R

```diff
@@ -24,46 +24,213 @@ if(pull_fresh_dataset) {
 
 
+####  Analysis  ####
 
-####  Data Cleaning  ####
-## Create summary variables
-clean_data <- qualtrics_data %>%
-  mutate(b_fas_sum = b_fas_1 + b_fas_2 + b_fas_3 + b_fas_4 + b_fas_5 + b_fas_6 + b_fas_7,
-         pi_fas_sum = pi_fas_1 + pi_fas_2 + pi_fas_3 + pi_fas_4 + pi_fas_5 + pi_fas_6 + pi_fas_7)
+##Duration
+
+names_in_order <- qualtrics_data %>%
+  select(ends_with("Page Submit")) %>%
+  names()
+
+#Put mean duration for each page into a new data frame
+page_durations <- qualtrics_data %>% #change to b_pi_completers dataset? If I do, I get a NaN result for line 120 and I'm not sure why :/
+  select(ends_with("Page Submit")) %>%
+  pivot_longer(cols = everything(),
+               names_to = "page",
+               values_to = "duration") %>%
+  group_by(page) %>%
+  summarise(mean_duration = mean(duration, na.rm = T),
+            sd_duration = sd(duration,  na.rm = T)) #seems odd/incorrect that the values I'm getting for standard deviation are so giant
+
+page_durations$page <- factor(page_durations$page, levels = names_in_order)
+
+#Plot mean duration for each page
+ggplot(data = page_durations) +
+  geom_col(aes(x = page, y = mean_duration)) +
+  coord_flip()
+
+#Identify mean duration for intro and pre-SSI questionnaires, the SSI, and post-SSI questionnaires
+b_ssi_pi_mean_durations <- page_durations %>% 
+  pivot_wider(names_from = page, values_from = mean_duration) %>% 
+  rename_with(~str_replace(.,'tim_Page Submit', 'page_submit')) %>% #didn't need this but leaving for my future reference
+  mutate(b_mean_duration_sec = sum(across(starts_with("b_")))) %>%  
+  mutate(b_mean_duration_minute = b_mean_duration_sec / 60) %>% 
+  mutate(ssi_mean_duration_sec = sum(across(starts_with("bn_")))) %>% #how to account for the fact that not every single person saw every page (e.g., the multiple story options)?
+  mutate(ssi_mean_duration_minute = ssi_mean_duration_sec / 60) %>%
+  mutate(pi_mean_duration_sec = sum(across(starts_with("pi_")))) %>% 
+  mutate(pi_mean_duration_minute = pi_mean_duration_sec / 60)
+
+b_ssi_pi_mean_durations$b_mean_duration_minute #4.112764 minutes
+b_ssi_pi_mean_durations$ssi_mean_duration_minute #27.76778 minutes
+b_ssi_pi_mean_durations$pi_mean_duration_minute #3.57133 minutes
+                        
+#should we also do standard deviations? If so, what's the best way? 
+
+##Program Feedback Scale
+
+#Total score (items summed) - how to better explain the difference between this and the one below?
+mean(pfs_completers$pi_pfs_sum)
+sd(pfs_completers$pi_pfs_sum)
+
+#Overall score (items not summed) - how to better explain the difference between this and the one above?
+pi_pfs_all_items <- data.frame(pfs_completers$pi_pfs_1, pfs_completers$pi_pfs_2, pfs_completers$pi_pfs_3,
+                            pfs_completers$pi_pfs_4, pfs_completers$pi_pfs_5, pfs_completers$pi_pfs_6, 
+                            pfs_completers$pi_pfs_7)
+mean(as.matrix(pi_pfs_all_items))
+sd(as.matrix(pi_pfs_all_items))
```

Comment:
Right now, this standard deviation is the standard deviation of all items. If you want the standard deviation of each person's mean score, you'd have to calculate each person's mean score first

Reply:
Unless you think this would be useful, I think I'll just scrap this section since we said "we will report means and standard deviations of the Program Feedback Scale, for each item and for overall score" which we do above and below this chunk.

Reply:
Sounds good